R/data-process.R

```diff
@@ -1,229 +1,333 @@
-              prevalence_diabetes = mean(data$prevalence_diabetes, na.rm = TRUE),
-              incidence_diabetes = mean(data$incidence_diabetes, na.rm = TRUE),
-              prevalence_intracerebralhemorrhage = mean(data$prevalence_intracerebralhemorrhage, na.rm = TRUE),
-              incidence_intracerebralhemorrhage = mean(data$incidence_intracerebralhemorrhage, na.rm = TRUE),
-              prevalence_ischemicstroke = mean(data$prevalence_ischemicstroke, na.rm = TRUE),
-              incidence_ischemicstroke = mean(data$incidence_ischemicstroke, na.rm = TRUE),
-              prevalence_asthma = mean(data$prevalence_asthma, na.rm = TRUE),
-              incidence_asthma = mean(data$incidence_asthma, na.rm = TRUE),
-              #prevalence_stomachcancer = mean(data$prevalence_stomachcancer, na.rm = TRUE),
-              #incidence_stomachcancer = mean(data$incidence_stomachcancer, na.rm = TRUE),
-              prevalence_subarachnoidhemorrhage = mean(data$prevalence_subarachnoidhemorrhage, na.rm = TRUE),
-              incidence_subarachnoidhemorrhage = mean(data$incidence_subarachnoidhemorrhage, na.rm = TRUE),
-              prevalence_stroke = prevalence_intracerebralhemorrhage + prevalence_ischemicstroke + prevalence_subarachnoidhemorrhage,
-              incidence_stroke = incidence_intracerebralhemorrhage + incidence_ischemicstroke + incidence_subarachnoidhemorrhage,
-              prevalence_ckd = mean(data$prevalence_chronickidneydisease, na.rm = TRUE),
-              incidence_ckd = mean(data$incidence_chronickidneydisease, na.rm = TRUE),
-              disability_weight = mean(data$disability_weight, na.rm = TRUE),
-              deaths = mean(data$deaths, na.rm = TRUE),
-              migrations = mean(data$migrations, na.rm = TRUE),
-              normal_weight = mean(data$normal_weight, na.rm = TRUE),
-              over_weight = mean(data$over_weight, na.rm = TRUE),
-              obese_weight = mean(data$obese_weight, na.rm = TRUE),
-              above_weight = mean(data$above_weight, na.rm = TRUE),
-              yll = mean(data$yll, na.rm = TRUE),
-              yld = mean(data$yld, na.rm = TRUE),
-              daly = mean(data$daly, na.rm = TRUE))
-
-  return(data_mean)
-}
-
-# Define global variables to suppress notes
-utils::globalVariables(c("incidence_intracerebralhemorrhage", "incidence_ischemicstroke",
-"incidence_subarachnoidhemorrhage", "prevalence_intracerebralhemorrhage",
-"prevalence_ischemicstroke", "prevalence_subarachnoidhemorrhage"))
+#' ## Step-by-Step Usage
+#'
+#' 1. **Read the data**: This function reads the data from the location specified.
+#'    ```r
+#'    data <-  readRDS("data.rds")
+#'    ```
+#'
+#' 2. **`gen_data_mean`**: Calculates weighted mean values for various metrics over years.
+#'    ```r
+#'    data_weighted <- gen_data_mean(data)
+#'    ```
+#'
+#' 3. **`gen_data_weighted_rf`**: Calculates the differences between intervention and baseline values for various metrics.
```

Comment:
Calculates the differences between intervention and baseline values for **risk factors**.